### ISSUE TITLE
Download all comments at once

### ISSUE THREAD

**Issue by Keats** [hidroh/materialistic#659]
The most annoying thing so far when using this app is that it only downloads a subset of the comments and more appear while scrolling.
Having an option to always download all the comments at once would be very nice, especially when loading an article before going underground etc

Very good app otherwise!

**Comment by hidroh** [comment]
Currently you can enable Offline download of comments & save the stories, which will pre-load everything and achieve the same thing you need here. I'll try to come up with a more straightforward option for such use case. Related to #655

**Comment by kilroythe5th** [comment]
even when I click the save button it still doesn't seem to save them all. maybe i'm doing it wrong, though

**Comment by hidroh** [comment]
@kilroythe5th were you on wifi when you saved?

**Comment by kilroythe5th** [comment]
I was connercted to wifi at the time, and it showed a progress bar of it downloading, but when I have tried to look at the article offline, it gives a "webpage not available". It does save all the comments, though.

**Comment by hidroh** [comment]
I'll investigate it. Does it happen to all articles or only a few selective ones?

**Comment by hidroh** [comment]
For the original request, one solution is to allow an option to switch to Algolia API, which allows downloading all descendants at once.

**Comment by hidroh** [comment]
Algolia API seems to have a different ordering of comments though, e.g. [Algolia](https://hn.algolia.com/api/v1/items/12733024) vs [original HN thread](https://news.ycombinator.com/item?id=12733024)

**Comment by kilroythe5th** [comment]
this seems to happen on all arcticles I tried. Can the app use the other API right now, or is it a feature you ar considering adding?

**Comment by GiuseppeArcuti** [comment]
@hidroh I would love this feature. I know some Java, if you could point me to the class where Comment fetching/saving is implemented I could try implementing.

**Comment by hidroh** [comment]
It's a bit more complicated than that i'm afraid. I'm still trying to find a way to make current code work.

**Comment by GiuseppeArcuti** [comment]
@hidroh Sorry I don't know the difficulties. Having used the app, it seems to me that when you request the comments the API returns a certain depth and as soon as you reach a comment you make a API request for a level deeper. So couldn't you just do a "virtual" scroll through the comments and request the deeper lever for each comment?

**Comment by hidroh** [comment]
That's not a bad idea, we can follow the same logic to quickly show the page like right now, and while user is still busy reading it, we preload the remaining content for them.

**Comment by GiuseppeArcuti** [comment]
@hidroh I was thinking more of a toggle in the settings. If you save a thread it downloads all the comments. Users who want more speed can toggle it off so it works like now.

**Comment by hidroh** [comment]
Yep new feature should always come with an optional setting to turn it on :) I don't want to upset existing users that are used to existing behaviors.

**Comment by GiuseppeArcuti** [comment]
Are you working on it or should I take a try? It takes so much time every
morning going through all the comments to download them.

2017-04-12 20:07 GMT+02:00 Ha Duy Trung <notifications@github.com>:

**Comment by hidroh** [comment]
For now you can save them, they will be scheduled for download (if you have offline comments enabled). I'm working on a feature that auto-download stories if you use home screen widget. Still working on it as it's quite battery heavy.

**Comment by nitrocode** [comment]
Hi @hidroh any update on this feature? Id love to see the new api implemented so we can search comments. Thank you for a great app by the way.
